File: src/main/node/magma/path/PathLike.ts
import Set from "../../java/util/Set";
import Result from "../result/Result";
import Option from "../option/Option";
/**
 * Minimal abstraction over file system paths. This wrapper lets the
 * rest of the code avoid a hard dependency on {@code java.nio.file.Path}.
 */
export interface PathLike {
    PathLike resolve(String other);
    PathLike relativize(PathLike other);
    PathLike getParent();
    Result<Set<PathLike>> walk();
    Result<String> readString();
    Option<String> createDirectories();
    Option<String> writeString(String text);
    Option<String> deleteIfExists();
    @Override String toString();
}

File: src/java/magma/api/io/PathLike.java
package magma.api.io;

import magma.api.list.Streamable;
import magma.api.option.Option;
import magma.api.result.Result;

public interface PathLike {
    Result<String, IOError> readString();

    Result<Streamable<PathLike>, IOError> walk();

    Option<IOError> writeString(CharSequence output);

    PathLike getFileName();

    String asString();
}
